apps/passport-server/src/services/userService.ts

```diff
@@ -287,6 +320,77 @@ export class UserService {
 
     return user;
   }
+
+  /**
+   * Updates the version of the legal terms the user agrees to
+   */
+  public async handleAgreeTerms(
+    serializedPCD: SerializedPCD<SemaphoreSignaturePCD>
+  ): Promise<AgreeTermsResult> {
+    const pcd = await SemaphoreSignaturePCDPackage.deserialize(
+      serializedPCD.pcd
+    );
+    if (await SemaphoreSignaturePCDPackage.verify(pcd)) {
```

Comment:
an early return in the case that the pcd doesn't verify would make this function more readable to me.